lib/GHCup/Download.hs

```diff
@@ -325,6 +327,45 @@ getDownloadInfo' t v = do
       _            -> with_distro <|> without_distro_ver <|> without_distro
     )
 
+getStackDownloadInfo :: ( MonadReader env m
+                        , HasDirs env
+                        , HasGHCupInfo env
+                        , HasLog env
+                        , HasPlatformReq env
+                        , HasSettings env
+                        , MonadCatch m
+                        , MonadFail m
+                        , MonadIO m
+                        , MonadMask m
+                        , MonadThrow m
+                        )
+                     => StackSetupURLSource
+                     -> [String]
+                     -> Tool
+                     -> GHCTargetVersion
+                     -- ^ tool version
+                     -> Excepts
+                          '[NoDownload, DownloadFailed]
+                          m
+                          DownloadInfo
+getStackDownloadInfo stackSetupSource keys@(_:_) GHC tv@(GHCTargetVersion Nothing v) =
+  case stackSetupSource of
+    StackSetupURL -> do
+      (dli :: Stack.SetupInfo) <- liftE $ reThrowAll @_ @_ @'[DownloadFailed] DownloadFailed $ getBase stackSetupURL
+      let siGHCs = Stack.siGHCs $ dli
+          ghcVersionsPerKey = (\key -> M.lookup key siGHCs) <$> (T.pack <$> keys)
+      ghcVersions <- (listToMaybe . catMaybes $ ghcVersionsPerKey) ?? (NoDownload tv GHC Nothing)
```

Comment:
Note: we pick the first key that has a valid version. We don't attempt multiple downloads.